### PR TITLE
refactor(state_table): remove pk in state_table insert_row/delete_row

### PR DESCRIPTION
**Pull request by wcy-fdu**
## What's changed and what's your intention?
As title.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by BowenXiao1999 — APPROVED** [review]

File: src/storage/src/table/state_table.rs [hunk 1]
```diff
@@ -132,7 +134,9 @@ impl<S: StateStore> StateTable<S> {
 
     /// This function scans rows from the relational table.
     pub async fn iter(&self, epoch: u64) -> StorageResult<impl RowStream<'_>> {
-        let cell_based_bounds = (Unbounded, Unbounded);
+        let cell_based_start_key = Included(self.keyspace.key().to_vec());
+        let cell_based_end_key = Excluded(next_key(self.keyspace.key()));
+        let cell_based_bounds = (cell_based_start_key, cell_based_end_key);
```
Comment (BowenXiao1999) [inline src/storage/src/table/state_table.rs:139]: Was this be fixed in another PR?

Comment (wcy-fdu) [reply src/storage/src/table/state_table.rs:139]: The former is merged.
